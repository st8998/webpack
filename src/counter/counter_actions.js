export const INC = 'COUNTER.INC'
export const inc = () => ({ type: INC })
