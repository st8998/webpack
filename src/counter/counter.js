import React, { Component, PropTypes } from 'react'

import { connect } from 'react-redux'
import { inc as incCounter } from './counter_actions'

export class Counter extends Component {
  static propTypes = {
    counter: PropTypes.number.isRequired,
    incCounter: PropTypes.func.isRequired
  }

  componentWillMount() {
    this.timer = window.setInterval(this.props.incCounter, 1000)
  }

  componentWillUnmount() {
    window.clearInterval(this.timer)
  }

  render() {
    return <span>Counted so far: <span style={{ color: 'orangered' }}>{this.props.counter}</span></span>
  }
}

export default connect(
  state => ({ counter: state.counter }),
  { incCounter }
)(Counter)
