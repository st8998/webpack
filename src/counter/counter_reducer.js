import { INC } from './counter_actions'

export default function counterReducer(state = 0, action) {
  switch (action.type) {
    case INC:
      return state + 1

    default:
      return state
  }
}
