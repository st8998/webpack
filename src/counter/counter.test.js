import React from 'react'
import { Counter } from './counter'
import renderer from 'react-test-renderer'

jest.useFakeTimers()

test('it renders counter', function () {
  const incMock = jest.fn()
  const counter = renderer.create(<Counter incCounter={incMock} counter={0} />)

  let tree = counter.toJSON()
  expect(tree).toMatchSnapshot()

  jest.runTimersToTime(1000)
  expect(incMock.mock.calls.length).toBe(1)
})
