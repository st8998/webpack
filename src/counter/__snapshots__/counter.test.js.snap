exports[`test it renders counter 1`] = `
<span>
  Counted so far: 
  <span
    style={
      Object {
        "color": "orangered",
      }
    }>
    0
  </span>
</span>
`;
